cmd/collectors/restperf/restperf.go

```diff
@@ -0,0 +1,760 @@
+								Msg("Set value failed")
+							continue
+						} else {
+							r.Logger.Trace().
+								Str("name", name).
+								Str("label", label).
+								Str("value", values[i]).
+								Msg("Set name.label = value")
+							count++
+						}
+					}
+				} else {
+					metr, ok := newData.GetMetrics()[name]
+					if !ok {
+						if metr, err = newData.NewMetricFloat64(name); err != nil {
+							r.Logger.Error().Err(err).
+								Str("name", name).
+								Msg("NewMetricFloat64")
+						}
+					}
+					metr.SetName(metric.label)
+					if c, err := strconv.ParseFloat(f.value, 64); err == nil {
+						if err = metr.SetValueFloat64(instance, c); err != nil {
+							r.Logger.Error().Err(err).Str("key", metric.name).Str("metric", metric.label).
+								Msg("Unable to set float key on metric")
+						}
+					} else {
+						r.Logger.Error().Err(err).Str("key", metric.name).Str("metric", metric.label).
+							Msg("Unable to set float key on metric")
+					}
+					count++
+				}
+			}
+		}
+		if err = newData.GetMetric("timestamp").SetValueFloat64(instance, ts); err != nil {
+			r.Logger.Error().Stack().Err(err).Msg("set timestamp value: ")
+		}
+		return true
+	})
+
+	if err != nil {
+		r.Logger.Error().Err(err).Msg("Error while processing end points")
+	}
+
+	r.Logger.Info().
+		Uint64("dataPoints", count).
+		Str("apiTime", apiD.String()).
+		Str("parseTime", parseD.String()).
+		Msg("Collected")
+
+	_ = r.Metadata.LazySetValueInt64("api_time", "data", apiD.Microseconds())
+	_ = r.Metadata.LazySetValueInt64("parse_time", "data", parseD.Microseconds())
+	_ = r.Metadata.LazySetValueUint64("count", "data", count)
+	r.AddCollectCount(count)
+
+	// skip calculating from delta if no data from previous poll
+	if r.prop.isCacheEmpty {
+		r.Logger.Debug().Msg("skip postprocessing until next poll (previous cache empty)")
+		r.Matrix = newData
+		r.prop.isCacheEmpty = false
+		return nil, nil
+	}
+
+	calcStart := time.Now()
+
+	r.Logger.Debug().Msg("starting delta calculations from previous cache")
+
+	// cache raw data for next poll
+	cachedData := newData.Clone(true, true, true)
+	// order metrics, such that those requiring base counters are processed last
+	orderedMetrics := make([]matrix.Metric, 0, len(newData.GetMetrics()))
+	orderedKeys := make([]string, 0, len(orderedMetrics))
+
+	for key, metric := range newData.GetMetrics() {
+		if metric.GetName() != "timestamp" {
+			counter := r.prop.counterInfo[metric.GetName()]
+			if counter == nil {
+				// check if it is an array counter
+				counter = r.prop.counterInfo[metric.GetComment()]
+			}
+			if counter != nil {
+				if counter.denominator == "" {
+					// does not require base counter
+					orderedMetrics = append(orderedMetrics, metric)
+					orderedKeys = append(orderedKeys, key)
+				}
+			} else {
+				r.Logger.Warn().Str("counter", metric.GetName()).Msg("Misconfiguration in template counter ")
+			}
+		}
+	}
+
+	for key, metric := range newData.GetMetrics() {
```

Comment:
wonder if this block and the very similar one above can be eliminated by adding metrics and sorting using a sort function that counter. denominator is blank before not blank denominator